drivers/dma/dma_stm32.c

```diff
@@ -594,8 +594,8 @@ static int dma_stm32_init(const struct device *dev)
 	config->config_irq(dev);
 
 #ifdef CONFIG_DMAMUX_STM32
-	int offset = ((dev == device_get_binding((const char *)"DMA_1"))
-			? 0 : config->max_streams);
+	int offset = (strncmp(dev->name, "DMA_1", 5)
```

Comment:
This is a little fragile, as the actual device name is from `DT_INST_LABEL(index)`.  If there's an instance-specific constraint on available streams that should probably come from devicetree.

Reply:
I agree that getting this piece of info from dts would be safer (in a next iteration probably).
@FRASTM would it be possible to add a relationship between DMA_X and DMAMUX within dmamux node bindings ?

Reply:
yes, 
adding a phandle array to the dmamux deviceTree named **dma-masters**

> dma-masters = <&dma1 &dma2>;


like the https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/plain/Bindings/dma/dma-router.yaml

Reply:
> yes, adding in a new further PR
> --> phandles to the dmamux deviceTree named **dma-masters**
> 
> > dma-masters = <&dma1 &dma2>;
> 
> like the https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/plain/Bindings/dma/dma-router.yaml

Looks good to me.
Then, what do we do with current PR ?
@pabigot do we use current fix for 2.4.0 and wait V2.5.0 for the DT version?

Reply:
If you don't think it can be addressed immediately then I'm OK with this as long as somebody opens an issue to ensure the cleanup is done for the next release.  Let me know when that's done.

Reply:
@pabigot a long term solution is being reviewed here: https://github.com/zephyrproject-rtos/zephyr/pull/28268.
This is not finalized yet.